awx/main/dispatch/worker/callback.py

```diff
@@ -152,7 +153,8 @@ def perform_work(self, body):
                     return
 
                 event = cls.create_from_data(**body)
-                self.buff.setdefault(cls, []).append(event)
+                if not event.skip_save:
+                    self.buff.setdefault(cls, []).append(event)
 
```

Comment:
What will the final set of counters look like for jobs that have events that are ignored with this toggle? Will they still be continuous? T

Reply:
Also, we have this mechanism in the API called "event_processing_finished" where it compares "what runner emitted" with "what got saved", and we use that to determine if event processing it _actually done_.  It seems like this PR would break that expectation.

Reply:
https://github.com/ansible/awx/blob/devel/awx/main/tasks.py#L1497
https://github.com/ansible/awx/blob/devel/awx/main/models/unified_jobs.py#L1004

Reply:
For this to all work properly, we'd need code here that *conditionally* incremented the counter:

https://github.com/ansible/awx/blob/devel/awx/main/tasks.py#L1234

Reply:
(we'd also need to verify that this worked properly with isolated nodes)

Reply:
@matburt what is Tower's event callback:

1.  Had code that looked at the event and determined **if** we should save it, and put a special key into the events that should be skipped.
2.  If skip=True, **don't** increment `emitted_events`.
3.  On the callback receiver side, `.pop('special_skip_key')`, and don't persist the event if the flag exists.

Reply:
related: https://github.com/ansible/awx/pull/6645

Reply:
I think the issue about save counts is now resolved. There seems to be a UI race condition now, though.

Reply:
I've shifted this code up to this event data entrypoint, I had to modify the exposure of the event level up to class scope.

Reply:
Something I noticed when looking at data on the websocket event bus - The `final_counter` on the EOF event isn't matching up with the stats event. No UI impact but probably worth looking into.

Reply:
I think this makes sense. The stats event is counting the tasks that come through Ansible. The `final_counter` is counted by the event processor in Tower itself which occurs during filtering of the events.

Reply:
I was unclear. I'm referring to the `counter` on the job_event that contains the `playbook_on_stats` data, not what the `playbook_on_stats` data is counting.

![Screenshot from 2020-04-09 16-20-14](https://user-images.githubusercontent.com/9753817/78937287-3b71c080-7a7e-11ea-9dfb-a21c49ce9d3e.png)

The EOF `final_counter` should match the highest job_event counter of any job_event emitted by the job. If you change the new setting to any value but `all` and run a job, the EOF `final_counter` doesn't match the highest job_event counter received by the client.

![Screenshot from 2020-04-09 16-20-34](https://user-images.githubusercontent.com/9753817/78937298-3f9dde00-7a7e-11ea-8419-1ef64bf433de.png)